lib/shellable.ts

```diff
@@ -50,6 +50,74 @@ export interface ShellableOptions {
    * @default a name is generated by CloudFormation.
    */
   buildProjectName?: string;
+
+  /**
+   * Can be used to run this build using a specific IAM role. This can be used,
+   * for example, to execute in the context of another account (e.g. to run
+   * tests in isolation).
+   */
+  assumeRole?: AssumeRole;
+}
+
+/**
+ * Properties used to create a Shellable
+ */
+export interface ShellableProps extends ShellableOptions {
+  /**
+   * Directory with the scripts.
+   *
+   * The whole directory will be uploaded.
+   */
+  scriptDirectory: string;
+
+  /**
+   * Filename of the initial script to start, relative to scriptDirectory.
+   */
+  entrypoint: string;
+}
+
+export interface AssumeRole {
+  /**
+   * The Amazon Resource Name (ARN) of the role to assume.
+   */
+  roleArn: string;
+
+  /**
+   * An identifier for the assumed role session.
+   *
+   * Use  the  role  session name to uniquely identify a session when the same
+   * role is assumed by different principals or for different reasons. In
+   * cross-account scenarios, the role session name is visible to, and can be
+   * logged by the account that owns the role.  The role session name is also
+   * used in the ARN of the assumed role principal. This means that subsequent
+   * cross-account API requests using the tem- porary security credentials will
+   * expose the role session name to the external account in their CloudTrail
+   * logs.
+   *
+   * The regex used to validate this parameter is a string of characters
+   * consisting  of upper- and lower-case alphanumeric characters with no
+   * spaces. You can also include underscores or any of the following
+   * characters: =,.@-
```

Comment:
Provide the regex?

Reply:
Copied this from AWS docs... Rephrasing. Why would the user care that there's a regex validating their input?